common/wireguard/src/wg_tunnel.rs

```diff
@@ -117,13 +113,17 @@ impl WireGuardTunnel {
             wg_tunnel,
             close_tx,
             close_rx,
-            tun_task_tx: tunnel_tx,
+            packet_tx,
             tag,
         };
 
         (tunnel, peer_tx, tag)
     }
 
+    fn new_tag() -> u64 {
+        rand::thread_rng().next_u64()
```

Comment:
Although small, there's still a possibility for collisions here, right? Maybe at least have some TODO comment for now reminding to check for tag collisions after generating a random one

Reply:
Yep good point